boa/src/builtins/map/mod.rs

```diff
@@ -248,7 +249,17 @@ impl Map {
 
         let size = if let Some(object) = this.as_object() {
             if let Some(map) = object.borrow_mut().as_map_mut() {
-                map.insert(key.clone(), value.clone());
+                let key = match key {
+                    JsValue::Rational(r) => {
+                        if r.is_zero() {
+                            JsValue::Rational(0f64)
+                        } else {
+                            key.into()
```

Comment:
```suggestion
                            key.clone()
```
Prefer using `clone` instead of `into` when converting a `&JsValue` into a `JsValue`. Also applies for the other `into`s.
We should remove the implementation of `From<&JsValue>` for `JsValue`, really.